frontend/app/ui_components/upload-progress-directive.js

```diff
@@ -0,0 +1,58 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+module.exports = function() {
+  var uploadProgressController = function(scope) {
+
+    scope.upload.progress(function(details) {
+      scope.file = details.config.file.name;
+      if (details.lengthComputable) {
+        scope.value = Math.round(details.loaded / details.total * 100);
+      } else {
+        // dummy value if not computable
+        scope.value = 10;
```

Comment:
Is there something like an indeterminate state for the progress control?

E.g. in .NET a progress bar can be determinate (indicating the progress based on value and max value) or indeterminate, showing a moving progress bar that can't be used to tell anything about the progress, but shows that "something is going on".